### PR TITLE
Expose TLS connection state when possible

### PR DESCRIPTION
**Pull request by ripienaar**
I would like to be able to report things like ciphers,
tls or not, verified or not etc in nats account info
where its shows other connection properties but had
no way to get at this information

Signed-off-by: R.I.Pienaar <rip@devco.net>

### CODE REVIEW

File: nats.go [hunk 1]
```diff
@@ -1807,6 +1808,24 @@ func (nc *Conn) makeTLSConn() error {
 	return nil
 }
 
+// TLSConnectionState retrieves the state of the TLS connection to the server
+func (nc *Conn) TLSConnectionState() (tls.ConnectionState, error) {
```
Comment (derekcollison) [inline nats.go:1812]: Should this be passed by value here? Is that the recommended approach?

Comment (derekcollison) [reply nats.go:1812]: Just looked seems so.

**Review by derekcollison — APPROVED** [review]
LGTM

File: nats.go (hunk 1 above)
Comment (ripienaar) [reply nats.go:1812]: yeah i find it awkward, i considered calling the go internal and returning a pointer which seemed a bit more idiomatic. So I am not sure if this is best, happy to do that pointer dance if you prefer

Comment (derekcollison) [reply nats.go:1812]: Their API does same so we are good IMO.

**Pull request merged**